public/pages/workflow_detail/workspace/resizable_workspace.tsx

```diff
@@ -107,22 +115,35 @@ export function ResizableWorkspace(props: ResizableWorkspaceProps) {
     }
   }
 
-  // Hook to update the workflow's flow state, if applicable. It may not exist if
-  // it is a backend-only-created workflow, or a new, unsaved workflow. If so,
-  // generate a default one based on the 'workflows' JSON field.
+  // Hook to update some default values for the workflow, if applicable. Flow state
+  // may not exist if it is a backend-only-created workflow, or a new, unsaved workflow.
+  // Metadata fields (name/description/use_case/etc.) may not exist if the user
+  // cold reloads the page on a new, unsaved workflow.
   useEffect(() => {
-    const workflowCopy = { ...props.workflow } as Workflow;
-    if (workflowCopy) {
-      if (!workflowCopy.workspaceFlowState) {
-        workflowCopy.workspaceFlowState = toWorkspaceFlow(
-          workflowCopy.workflows
-        );
-        console.debug(
-          `There is no saved UI flow for workflow: ${workflowCopy.name}. Generating a default one.`
-        );
-      }
-      setWorkflow(workflowCopy);
+    let workflowCopy = { ...props.workflow } as Workflow;
+    if (!workflowCopy.ui_metadata || !workflowCopy.ui_metadata.workspaceFlow) {
+      workflowCopy.ui_metadata = {
+        ...(workflowCopy.ui_metadata || {}),
+        workspaceFlow: toWorkspaceFlow(workflowCopy.workflows),
+      };
+      console.debug(
+        `There is no saved UI flow for workflow: ${workflowCopy.name}. Generating a default one.`
+      );
```

Comment:
Do we just do a debug or some alert here as well?

Reply:
I'm keeping as just a logged debug for now. It's not considered an error as it would just indicate users loading an existing workflow with no ui metadata.